apps/challenges/challenge_config_utils.py

```diff
@@ -220,436 +239,757 @@ def get_value_from_field(data, base_location, field_name):
+                self.challenge_config_location,
+            )
+            if not is_valid:
+                self.error_messages.append(message)
+            else:
+                self.yaml_file_data["submission_guidelines"] = get_value_from_field(
+                    self.yaml_file_data,
+                    self.challenge_config_location,
+                    "submission_guidelines",
+                )
 
-    # Check for evaluation script path
-    evaluation_script = yaml_file_data.get("evaluation_script")
-    if evaluation_script:
-        evaluation_script_path = join(
-            challenge_config_location, evaluation_script
-        )
-        # Check for evaluation script file in extracted zip folder
-        if isfile(evaluation_script_path):
-            challenge_evaluation_script_file = read_file_data_as_content_file(
-                evaluation_script_path, "rb", evaluation_script_path
+    def validate_evaluation_script_file(self):
+        evaluation_script = self.yaml_file_data.get("evaluation_script")
+        if evaluation_script:
+            evaluation_script_path = join(
+                self.challenge_config_location, evaluation_script
             )
-            files[
-                "challenge_evaluation_script_file"
-            ] = challenge_evaluation_script_file
+            # Check for evaluation script file in extracted zip folder
+            if isfile(evaluation_script_path):
+                self.challenge_evaluation_script_file = read_file_data_as_content_file(
+                    evaluation_script_path, "rb", evaluation_script_path
+                )
+                self.files[
+                    "challenge_evaluation_script_file"
+                ] = self.challenge_evaluation_script_file
+            else:
+                message = self.error_messages_dict.get("missing_evaluation_script")
+                self.error_messages.append(message)
         else:
-            message = "ERROR: No evaluation script is present in the zip file. Please add it and then try again!"
-            error_messages.append(message)
-    else:
-        message = "ERROR: There is no key for evaluation script in YAML file. Please add it and then try again!"
-        error_messages.append(message)
-
-    if not len(error_messages):
-        serializer = ZipChallengeSerializer(
-            data=yaml_file_data,
-            context={
-                "request": request,
-                "challenge_host_team": challenge_host_team,
-                "image": challenge_image_file,
-                "evaluation_script": challenge_evaluation_script_file,
-                "github_repository": request.data["GITHUB_REPOSITORY"],
-            },
-        )
-        if not serializer.is_valid():
-            message = "ERROR: Challenge metadata has following schema errors:\n {}".format(
-                str(serializer.errors)
+            message = self.error_messages_dict.get("missing_evaluation_script")
```

Comment:
Both the errors are different here.
One is missing `evaluation_script` key. One is missing evaluation script file. You have used the same error here for both which is incorrect!

Reply:
message = "ERROR: No evaluation script is present in the zip file. Please add it and then try again!"

Reply:
message = "ERROR: There is no key for evaluation script in YAML file. Please add it and then try again!"